src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -1929,6 +1932,19 @@ JRT_ENTRY(void, Deoptimization::uncommon_trap_inner(JavaThread* current, jint tr
 
     JFR_ONLY(post_deoptimization_event(nm, trap_method(), trap_bci, trap_bc, reason, action);)
 
+    if(Log(deoptimization)::is_info()) {
```

Comment:
I don't see this pattern used for checking whether a log tag is enabled used elsewhere. Won't this only print when the log level is exactly `info`?

The pattern I see used elsewhere is:

```
LogTarget(Info, <tag>) lt;
if (lt.is_enabled()) {
    ...
}
```

Reply:
This does the same thing. `is_enabled` calls `is_level` internally. I did it this way so that we have a temporary value instead of cluttering the outer scope.

Reply:
Using a `LogMessage` instead of a `LogTarget` is very rarely used in the codebase. I have no idea what the pros and cons of each are, nor can I find any guidance on which to use when.

Reply:
Also style nit: space after if please.